utils/filter_streams.h

```diff
@@ -124,6 +135,19 @@ class filter_ostream : public std::ostream {
     std::map<std::string, std::string> options;
 };
 
+template<typename T>
+filter_ostream & operator << (filter_ostream & stream, T && data)
```

Comment:
Should be made a member function since there's no value in having it outside the class.